src/ddmd/statementsem.d

```diff
@@ -483,6 +490,366 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
         result = fs;
     }
 
+    private static template MakeTupleForeachRet(bool isDecl){
+        static if(isDecl)
+        {
+            alias MakeTupleForeachRet = Dsymbols*;
+        }
+        else
+        {
+            alias MakeTupleForeachRet = void;
+        }
+    }
+
+    MakeTupleForeachRet!isDecl makeTupleForeach(bool isStatic, bool isDecl)(ForeachStatement fs, TupleForeachArgs!(isStatic, isDecl) args)
```

Comment:
You're right, much of it is copy-pasta. But the new stuff should be documented. Poorly documented existing code is not a justification for more of such :-)